lib/handlers/authorize-handler.js

```diff
@@ -167,8 +165,7 @@ AuthorizeHandler.prototype.getClient = function(request) {
   if (redirectUri && !is.uri(redirectUri)) {
     throw new InvalidRequestError('Invalid request: `redirect_uri` is not a valid URI');
   }
-
-  return Promise.try(this.model.getClient, clientId)
+  return Promise.try(promisify(this.model.getClient, 1), clientId)
```

Comment:
This should be `return Promise.try(promisify(this.model.getClient, 2), [clientId, null])`, as the signature of `getClient` is `(clientId, clientSecret, callback)`. Currently `promisify`'s callback is passed as `clientSecret`.

```
model.getClient = function getClient(clientId, clientSecret, callback) {
    let client = /* get client using clientId */;
    if (clientSecret) {
        // oops..
    }
    // ...
}
```

This could be prevented in the model by performing a check like `typeof clientSecret === 'function'` but callback support should really be optional. If I decide to use promises (or throw) I shouldn't have to be aware of this edge case.